branchpro/models.py

```diff
@@ -42,3 +43,119 @@ def simulate(self, parameters, times):
             value appears in ``times``.
         """
         raise NotImplementedError
+
+#
+# BranchProModel Class
+#
+
+
+class BranchProModel(ForwardModel):
+    r"""BranchProModel Class:
+    Class for the models following a Branching Processes behaviour.
+    It inherits from the `ForwardModel`` class.
+
+    In the branching process model, we track the number of cases
+    registered each day, I_t, also known as the "incidence" at time t.
+
+    The incidence at time t is a random variable distributed according to
+    a Poisson distribution with a mean that depends on previous number of
+    cases, according to the following formula:
+
+    .. math::
+        E(I_{t}^{\text(local)}|I_0, I_1, \dots I_{t-1}, w_{s}, R_{t}) =
+            R_{t}\sum_{s=1}^{t}I_{t-s}w_{s}
+
+    Parameters
+    ----------
+    initial_r
+        (numeric) Value of the reproduction number at the beginning
+        of the epidemic
+    serial_interval:
+        (list) Unnormalised probability distribution of that the recipient
+        first displays symptoms s days after the infector first displays
+        symptoms
+
+    Methods
+    -------
+    simulate: return model output for specified parameters and times.
+    _normalised_daily_mean: (Private) returns the expected number of new cases
+        at time t
+    """
+
+    def __init__(self, initial_r, serial_interval):
+        super(BranchProModel, self).__init__()
+
+        if not isinstance(serial_interval, (list, np.ndarray)):
```

Comment:
Could you possibly just remove this statement? The np.asarray statement below makes sure that serial_interval will be a numpy array. If the input cannot be convert to a numpy array this function will throw an error that is pretty understandable. So we don't need to check the type here

Reply:
I cannot, because if I do, then the message is actually weird:

I tried ``print(BranchProModel(0, 1).simulate(1,1))`` and this is what I got:

<img width="693" alt="Screenshot 2020-10-28 at 21 53 40" src="https://user-images.githubusercontent.com/72917474/97500801-24d27e80-1968-11eb-95bd-b4268f58e7b1.png">

Insisting on a list or Numpy array format solves that issue

Reply:
Ahhh I see! This problem would however also occur if we pass a serial_interval with dimension 0, i.e. if np.array(1). So what's probably best is to convert the serial_interval to a numpy array and than make an assert that the number of dimenions is 1, ie.e.

```
serial_interval = np.asarray(serial_interval)
if serial_interval.ndim != 1:
    raise ValueError('...')

self._serial_interval = serial_interval
```

Reply:
If you look at line 96, I have left that as it was even when deleting the raise error and the message if gave is the one that I captioned above.

Reply:
Okay, I have done as you suggested. Check final commit.